script.py

```diff
@@ -52,73 +52,117 @@ def check_data_frame(data_frame, spec):
             raise Exception("Unexpected data frame column: {0}".format(data_frame_column))
 
 
-def generate_all_html_pages():
+def generate_heatmap():
     stations = get_station_list()
     trips = add_lat_and_lon_to_trips(get_and_format_trip_data("resources/Divvy_Trips_2018_06.csv"), stations)
-    stations = add_trip_counts_to_stations(stations, trips)
-    station_map = put_stations_on_map(stations)
-
-    #makes timeseries heatmap
     timeseries_heatmap_data = create_timeseries_data(trips)
     make_timeseries_map(timeseries_heatmap_data)
 
-    #code below here makes map with path lines on it
+def create_html_map_with_path_data():
+    trips = get_format_path_data()
+    folium_map = create_folium_image_with_paths(trips)
+    generate_html(folium_map)
+
+def create_folium_image_with_paths(trips):
+    min_lat, max_lat, max_lon, min_lon = find_max_and_min_lat_and_lon(trips)
+    image_data = create_map_image_data(trips)
+    map_overlay = make_map_image_overlay(image_data)
+    delta_lat = get_aspect_ratio_and_return_delta_lat(map_overlay, max_lon, min_lon, min_lat)
+    folium_map = make_folium_map(map_overlay, max_lat, delta_lat, min_lon, max_lon)
+    return folium_map
+
+def get_format_path_data():
+    trips = get_trips()
+    trips = identify_discrete_trips(trips)
```

Comment:
I think we can chain all of these function calls and not have any of the functions return `trips`, since we are using pointers. I think it would make for cleaner looking code, let's discuss